src/scanoss/inspection/undeclared_component.py

```diff
@@ -115,23 +122,46 @@ def _markdown(self, components: list) -> Dict[str,Any]:
             'summary': self._get_summary(components),
         }
 
-    def _generate_sbom_file(self, components: list) -> dict:
+    def _get_unique_components(self, components: list) -> list:
         """
-         Generate a list of PURLs for the SBOM file.
+         Generate a list of unique components.
 
          :param components: List of undeclared components
-         :return: SBOM Dictionary with components
+         :return: list of unique components
          """
-
         unique_components = {}
         if components is None:
             self.print_stderr(f'WARNING: No components provided!')
```

Comment:
i think here for all warning messages we should use `self.print_debug` to only show them if debug mode is turned on

Reply:
It depends. If we expect entries, then it should be a warning. If it can be empty, it should be debug. So, which is it? :-)

Reply:
Agree :)